code/game/objects/items/sticker/sticker.dm

```diff
@@ -0,0 +1,153 @@
+/obj/item/sticker
+	name = "sticker" //Don't bother changing the name or desc for subtypes
+	desc = "An adhesive graphic."
+	icon = 'icons/obj/sticker.dmi'
+	icon_state = "happy"
+	flags_1 = IS_ONTOP_1
+	w_class = WEIGHT_CLASS_TINY
+	appearance_flags = TILE_BOUND | PIXEL_SCALE | KEEP_APART
+	///Our current state for being stuck or unstuck
+	var/sticker_state = STICKER_STATE_ITEM
+	///Built appearance for item state
+	var/mutable_appearance/item_appearance
+	///Build appearance for stuck state
+	var/mutable_appearance/stuck_appearance
+	///Sticker icon
+	var/sticker_icon
+	var/sticker_icon_state = "happy_sticker"
+	///Do we add an outline?
+	var/do_outline = TRUE
+	///Sticker flags
+	var/sticker_flags
+	///Drop rate weight, keep this seperate from rarity for joke
+	var/drop_rate = STICKER_WEIGHT_COMMON
+	///Do we roll for unusual effects
+	var/roll_unusual = TRUE
+	var/is_unusual = FALSE
+
+/obj/item/sticker/Initialize(mapload)
+	. = ..()
+	item_appearance = build_item_appearance()
+	stuck_appearance = build_stuck_appearance()
+	//Sticker outline
+	if(do_outline)
+		add_filter("sticker_outline", 1, outline_filter(1.3, "#fff", flags = OUTLINE_SHARP))
+	//Unusual stuff
+	if(roll_unusual)
+		generate_unusual()
+
+/obj/item/sticker/afterattack(atom/movable/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(!can_stick(target) || !proximity_flag)
+		return
+	//Move to our target
+	forceMove(target)
+	layer = target.layer+0.01
+	target.vis_contents += src
```

Comment:
Is this a vis contents thing to have clicks go to the sticker instead? Would it be better to use overlays and have alt-click be the thing that removes the sticker to prevent annoyances with inventory management?

Reply:
I'd rather not change it. The current implementation is very simple to understand and intuitive, you click to put on and click to take off. All non-help intent clicks will be copied over to the holder, this works with bullets and most other things.